### ISSUE TITLE
Passwords containing `:`

### ISSUE THREAD

**Issue by benel** [Hypertopic/AAAforREST#36]
... are not properly parsed.

This is due to the way HTTP basic authentication is parsed:
`token.split(':')[1]` is not necessarily the whole password.

**Comment by benel** [comment]
In [proxy.js](https://github.com/Hypertopic/AAAforREST/blob/master/app/proxy.js#L246-L247), try to replace:
```js
    login: credentials[0],
    password: credentials.length>1 ? credentials[1] : ""
```
with
```js
    login: credentials.shift(),
    password: credentials.length>0 ? credentials.join(':') : ""

**Comment by PunKeel** [comment]
Hi, I'm hit by this bug and can't login to Hyper Topic (Maxime Guerreiro)

**Comment by benel** [comment]
@PunKeel If you're speaking about your university LDAP password, it can be changed through the university digital work environment (ENT).

**Comment by PunKeel** [comment]
Why don't you just merge the code into master ?

(BTW there is no need for the length check, as `[].join(...)` is the empty string)

**Comment by benel** [comment]
> (BTW there is no need for the length check, as [].join(...) is the empty string)

True.

> Why don't you just merge the code into master ?

Because right now I don't have the time to test it so I prefer my colleagues to do it.

**Comment by benel** [comment]
Fixed by commit 729708e81e9034ba2681024da4f16a43e5d772a2.

@PunKeel Thank you very much for your nice contribution. 
Feel free to contact me for a student job or project (aka TX) on next semester.
